src/main/java/com/gooddata/http/client/GoodDataHttpClient.java

```diff
@@ -200,16 +225,20 @@ private HttpResponse handleResponse(final HttpHost httpHost, final HttpRequest r
      * </ul>
      * @throws GoodDataAuthException error
      */
-    private boolean refreshTt(final HttpHost httpHost) throws IOException {
+    private boolean refreshTt() throws IOException {
         log.debug("Obtaining TT");
+
         final HttpGet getTT = new HttpGet(TOKEN_URL);
+        getTT.addHeader(SST_HEADER, sst);
         try {
-            final HttpResponse response = httpClient.execute(httpHost, getTT, context);
+            final HttpResponse response = httpClient.execute(authHost, getTT, (HttpContext) null);
```

Comment:
couldn't the `execute(HttpHost host, HttpRequest req)` method be used here? Or is the default context somehow breaking things?